internal/lockclient/simple_client.go

```diff
@@ -103,3 +133,91 @@ func (sc *SimpleClient) Release(d lockservice.Descriptors) error {
 func (sc *SimpleClient) StartService(cfg Config) error {
 	panic("TODO")
 }
+
+// Checkacquire checks for acquisition of lock and returns the owner if the lock
+// is already acquired.
+// The errors returned can be due to HTTP errors or marshalling errors.
+// A "file is not acquired" error is returned if so and no error and an owner is
+// returned if the object is acquired.
+func (sc *SimpleClient) Checkacquire(d lockservice.ObjectDescriptor) (string, error) {
+	if sc.cache != nil {
+		owner, err := sc.getFromCache(d)
```

Comment:
can't `sc.getFromCache(d)` be returned straightaway? The next few lines seem to be a repetition of the code already present in that function

Reply:
yep good catch!
I believe there were some other checks before and this stayed because others were removed and I poverlooked